src/main/java/ecs/SpriteAnimation.java

```diff
@@ -33,11 +33,11 @@
     };
 
     public SpriteAnimation(SpriteRenderer renderer, Sprite baseImage, float timePerSprite) {
+        super(ComponentOrder.POST_DRAW);
```

Comment:
setting this to post draw may delay the animation by one frame.

Reply:
yes, but it ensure, that the sprite is set and the sprite rendering has a chance to update before the animation is applied to it

Reply:
and to be clear: post draw doesnt mean it gets updated after drawing the screen - its still within the update loop